### PR TITLE
add tests with postgres 12 (close #2749)

### LINKED ISSUE #2749: update pg_dump version bundled with server
the current one is 11, we should update it to the latest postgres release

### PR DESCRIPTION
**Pull request by shahidhk**
### Description

Updates the circle ci config to add tests with Postgres 12

Note: `pg_dump` should be updated to 12 before merging this.

### Affected components 

- Build System
- Tests

### CODE REVIEW

**Comment by shahidhk** [comment]
`pg_dump` should be updated to 12 before merging this.

**Comment by lexi-lambda** [comment]
@shahidhk I’ve pushed a patch to the server to fix the failing test.

File: .circleci/config.yml [hunk 1]
```diff
@@ -231,6 +231,15 @@ jobs:
         command: echo 'all server tests passed!'
 
   # pytest the server with postgres versions >= 9.5
+  pytest_server_pg_12:
```
[the remark this answers is not in the document]
Comment (shahidhk) [reply .circleci/config.yml]: Sure, it is 12_0, right?

[the remark this answers is not in the document]
Comment (shahidhk) [reply .circleci/config.yml]: We'll need to fix for all versions then, Let's do it as a separate PR?

**Pull request merged**